agent/registration/host_registrar.go

```diff
@@ -129,3 +141,37 @@ func (hr *HostRegistrar) GetNetworkStatus() []infrastructurev1beta1.NetworkStatu
 	}
 	return Network
 }
+
+// getHostInfo gets the host platform details...
+func (hr *HostRegistrar) getHostInfo() (infrastructurev1beta1.HostInfo, error) {
+	hostInfo := infrastructurev1beta1.HostInfo{}
+
+	hostInfo.Architecture = runtime.GOARCH
+	hostInfo.OSName = runtime.GOOS
+
+	if distribution, err := getOperatingSystem(); err != nil {
+		return hostInfo, errors.Wrap(err, "failed to get host operating system image")
+	} else {
+		hostInfo.OSImage = distribution
+	}
+	return hostInfo, nil
+}
+
+// getOperatingSystem gets the name of the current operating system image.
+func getOperatingSystem() (string, error) {
+	rex := regexp.MustCompile("(PRETTY_NAME)=(.*)")
+
+	bytes, err := ioutil.ReadFile("/etc/os-release")
+	if err != nil && os.IsNotExist(err) {
+		// /usr/lib/os-release in stateless systems like Clear Linux
+		bytes, err = ioutil.ReadFile("/usr/lib/os-release")
+	}
+	if err != nil {
+		return "", fmt.Errorf("error opening file : %v", err)
+	}
+	line := rex.FindAllStringSubmatch(string(bytes), -1)
+	if len(line) > 0 {
+		return strings.Trim(line[0][2], "\""), nil
+	}
+	return "Linux", nil
+}
```

Comment:
Could we add some tests for this?

Reply:
Yes but that would be platform dependent so if we change the test environment to some other operating system the test will fail.

Reply:
Isn't the above code platform agnostic for all the potential platforms for `BYOH`(Linux)?

Reply:
Okay, My idea is to primarily test all the regex expressions. Is it possible to mock `ioutil.ReadFile`?

Reply:
Yes. Done.